sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -207,13 +219,21 @@ class HadoopTableReader(
    * If `filterOpt` is defined, then it will be used to filter files from `path`. These files are
    * returned in a single, comma-separated string.
    */
-  private def applyFilterIfNeeded(path: Path, filterOpt: Option[PathFilter]): String = {
-    filterOpt match {
-      case Some(filter) =>
-        val fs = path.getFileSystem(sc.hiveconf)
-        val filteredFiles = fs.listStatus(path, filter).map(_.getPath.toString)
-        filteredFiles.mkString(",")
-      case None => path.toString
+  private def applyFilterIfNeeded(path: Path, filterOpt: Option[PathFilter]): Option[String] = {
+    if (fs.exists(path)) {
```

Comment:
I think we'd better get `fs` from the `path`,because in the `hadoop namenode federation` we may get some problems like ` Wrong FS` exception if we  use the `FileSystem.get(sc.hiveconf)`  to get fs.